File: generator/src/test/java/dev/hilla/generator/basic/ExternalEntitiesEndpoint.java
package dev.hilla.generator.basic;

import com.cromoteca.samples.Simple;
import dev.hilla.Endpoint;
import java.util.List;
import org.springframework.data.domain.Page;
import reactor.core.publisher.Flux;

@Endpoint
public class ExternalEntitiesEndpoint {
  public void page(Page<List<String>> argument) {}

  public void flux(Flux<List<String>> argument) {}

  public void inner(Inner<String> argument) {}

  public void simple(Simple argument) {}

  public static class Inner<T> {
    public Page<T> page;
    public Flux<T> flux;
  }
}

File: generator/src/test/resources/com/cromoteca/generator/basic/ExternalEntitiesEndpoint.ts
import client from '../../../../connect-client.default.js';
import type Page from '../../../../org/springframework/data/domain/Page.js';
import type Flux from '../../../../reactor/core/publisher/Flux.js';
import type Simple from '../../samples/Simple.js';
import type Inner from './ExternalEntitiesEndpoint/Inner.js';
import type { EndpointRequestInit } from '@hilla/frontend';

async function flux(argument: Flux<string[]>, init?: EndpointRequestInit): Promise<void> {
    return client.call('com.cromoteca.generator.basic.ExternalEntitiesEndpoint', 'flux', { argument }, init);
}

async function inner(argument: Inner<string>, init?: EndpointRequestInit): Promise<void> {
    return client.call('com.cromoteca.generator.basic.ExternalEntitiesEndpoint', 'inner', { argument }, init);
}

async function page(argument: Page<string[]>, init?: EndpointRequestInit): Promise<void> {
    return client.call('com.cromoteca.generator.basic.ExternalEntitiesEndpoint', 'page', { argument }, init);
}

async function simple(argument: Simple, init?: EndpointRequestInit): Promise<void> {
    return client.call('com.cromoteca.generator.basic.ExternalEntitiesEndpoint', 'simple', { argument }, init);
}

const ExternalEntitiesEndpoint = {
    flux,
    inner,
    page,
    simple,
};

export default ExternalEntitiesEndpoint;
